// ngx-extended-pdf-viewer/esm2020/lib/ngx-extended-pdf-viewer-server.service.mjs
export class NgxExtendedPdfViewerService {
    findMultiple(text, options = {}) {
        return false;
    }
    find(text, options = {}) {
        return false;
    }
    findNext() {
        return false;
    }
    findPrevious() {
        return false;
    }
    print(printRange) {
        return false;
    }
    removePrintRange() { }
    setPrintRange(printRange) { }
    filteredPageCount(pageCount, range) {
        return 0;
    }
    isInPDFPrintRange(pageIndex, printRange) { }
    async getPageAsText(pageNumber) {
        return '';
    }
    async getPageAsImage(pageNumber, scale, background, backgroundColorToReplace = '#FFFFFF') {
        return;
    }
    async getCurrentDocumentAsBlob() {
        return new Blob([], { type: 'application/pdf' });
    }
    async getFormData() {
        return [];
    }
    addPageToRenderQueue(pageIndex) {
        return false;
    }
    isRenderQueueEmpty() {
        return true;
    }
    hasPageBeenRendered(pageIndex) {
        return false;
    }
    numberOfPages() {
        return 0;
    }
    getCurrentlyVisiblePageNumbers() {
        return [];
    }
    recalculateSize() { }
    async listLayers() {
        return;
    }
    async toggleLayer(layerId) { }
    scrollPageIntoView(pageNumber, pageSpot) { }
}
