// ngx-extended-pdf-viewer/esm2020/lib/responsive-visibility.mjs
import { Pipe } from '@angular/core';
import * as i0 from "@angular/core";
export class PdfBreakpoints {
}
PdfBreakpoints.xs = 490;
PdfBreakpoints.sm = 560;
PdfBreakpoints.md = 610;
PdfBreakpoints.lg = 660;
PdfBreakpoints.xl = 740;
PdfBreakpoints.xxl = 830;
export class ResponsiveCSSClassPipe {
    transform(visible, defaultClass) {
        switch (visible) {
            case undefined:
                return defaultClass;
            case false:
                return 'invisible';
            case true:
                return defaultClass;
            case 'always-visible':
                return '';
            case 'xs':
                return 'hiddenTinyView';
            case 'sm':
                return 'hiddenSmallView';
            case 'md':
                return 'hiddenMediumView';
            case 'lg':
                return 'hiddenLargeView';
            case 'xl':
                return 'hiddenXLView';
            case 'xxl':
                return 'hiddenXXLView';
        }
    }
}
ResponsiveCSSClassPipe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.3.12", ngImport: i0, type: ResponsiveCSSClassPipe, deps: [], target: i0.ɵɵFactoryTarget.Pipe });
ResponsiveCSSClassPipe.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "12.0.0", version: "13.3.12", ngImport: i0, type: ResponsiveCSSClassPipe, name: "responsiveCSSClass" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.3.12", ngImport: i0, type: ResponsiveCSSClassPipe, decorators: [{
            type: Pipe,
            args: [{ name: 'responsiveCSSClass' }]
        }] });
export class NegativeResponsiveCSSClassPipe {
    transform(visible, defaultClass) {
        switch (visible) {
            case undefined:
                return 'invisible';
            case false:
                return 'invisible';
            case true:
                return defaultClass;
            case 'always-visible':
                return 'invisible';
            case 'xs':
                return 'visibleTinyView';
            case 'sm':
                return 'visibleSmallView';
            case 'md':
                return 'visibleMediumView';
            case 'lg':
                return 'visibleLargeView';
            case 'xl':
                return 'visibleXLView';
            case 'xxl':
                return 'visibleXXLView';
        }
    }
}
NegativeResponsiveCSSClassPipe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.3.12", ngImport: i0, type: NegativeResponsiveCSSClassPipe, deps: [], target: i0.ɵɵFactoryTarget.Pipe });
NegativeResponsiveCSSClassPipe.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "12.0.0", version: "13.3.12", ngImport: i0, type: NegativeResponsiveCSSClassPipe, name: "negativeResponsiveCSSClass" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.3.12", ngImport: i0, type: NegativeResponsiveCSSClassPipe, decorators: [{
            type: Pipe,
            args: [{ name: 'negativeResponsiveCSSClass' }]
        }] });
