// ngx-extended-pdf-viewer/esm2020/lib/toolbar/pdf-findbar/pdf-findbar-service.mjs
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
export class PdfFindbarService {
    constructor() {
        this.multipleSearchTexts = false;
        this._individualWordsMode = true;
    }
    get individualWordsMode() {
        return this._individualWordsMode;
    }
    set individualWordsMode(value) {
        if (this._individualWordsMode != value) {
            const multilineInput = document.querySelector('ngx-extended-pdf-viewer #findInputMultiline');
            const wordsInput = document.querySelector('ngx-extended-pdf-viewer #findInput');
            if (value) {
                const query = multilineInput.value;
                if (query) {
                    wordsInput.value = query.replace(/\n/, ' ');
                }
            }
            else {
                const query = wordsInput.value;
                if (query) {
                    multilineInput.value = query;
                }
            }
        }
        this._individualWordsMode = value;
        setTimeout(() => {
            const PDFViewerApplication = window.PDFViewerApplication;
            PDFViewerApplication.findBar.dispatchEvent('');
        });
    }
}
PdfFindbarService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.3.12", ngImport: i0, type: PdfFindbarService, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
PdfFindbarService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "13.3.12", ngImport: i0, type: PdfFindbarService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.3.12", ngImport: i0, type: PdfFindbarService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root',
                }]
        }] });
